### PR TITLE
Change default Size value in New-AzVM to Standard_D2S_v3

### PR DESCRIPTION
**Pull request by Sandido**
Design Spec: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/889

Dingmeng saw an increase in failure rates for the New-AzVM cmdlet in the PS telemetry. Adam investigated and saw that the Size parameter has a hardcoded default value of Standard_Ds1_v2. This Size can be made unavailable in certain regions due to changes in activity in that region. If it is unavailable, the New-AzVM cmdlet fails if it is not provided a Size value. This occurs in its DiskFile and Simple parameter sets. 

After a discussion with the Compute PMs, PS Lead Dingmeng, and CLI team, we determined the best solution for now is to change the hardcoded value to a more widely available size, Standard_D2s_v2. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - Azure/azure-powershell-cmdlet-review-pr#889
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by BethanyZhou** [comment]
Hi @Sandido , please re-record test cases for src\Compute\Compute.Test\ScenarioTests\VirtualMachineTests.cs

**Comment by BethanyZhou** [comment]
@Sandido , It seems like there are some errors in the VM test cases. Could you configure it out so that the CI can be passed. I guess re-recording test cases for src\Compute\Compute.Test\ScenarioTests\VirtualMachineTests.cs would be work.

**Comment by Sandido** [comment]
> @Sandido , It seems like there are some errors in the VM test cases. Could you configure it out so that the CI can be passed. I guess re-recording test cases for src\Compute\Compute.Test\ScenarioTests\VirtualMachineTests.cs would be work.

@BethanyZhou , looks like when I record the new test, it succeeds. But whenever I try to record a test then immediately running it in Playback, it fails bc it is unable to find a Get call on the Recorded file.

**Comment by BethanyZhou** [comment]
Have you copy the new generated json files under bin/SessionRecords folder to the SessionRecords folder in Compute.Test.csproj?

**Comment by Sandido** [comment]
Looks like I needed to provide a DomainNameLabel value in New-AzVM. It defaults to a value in Record, but then on Playback it defaults to another value. I added DomainNameLabel as a parameter to fix it.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 1]
```diff
@@ -4627,7 +4627,7 @@ function Test-NewAzVMDefaultingSize
 {
     # Setup
     $rgname = Get-ComputeTestResourceName;
-    $loc = Get-ComputeVMLocation;
+    $loc = "eastus";#Get-ComputeVMLocation;
```
Comment (BethanyZhou) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:4630]: Hi @Sandido , please remove uncommented codes.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 1 above)
Comment (Sandido) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:4630]: updated.

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Comment by Sandido** [comment]
@BethanyZhou , I'm confused. This PR had no errors to Master, and now it has errors against the release branch. The error in the PR check does not look related to my changes. 
Why is this rebased to the release branch? 

Current PR check error:
/home/vsts/work/1/s/build.proj(182,5): error MSB3073: The command "dotnet sln /home/vsts/work/1/s/artifacts/Azure.PowerShell.sln add "/home/vsts/work/1/s/src/ApplicationInsights/ApplicationInsights/ApplicationInsights.csproj"" exited with code 139.
2021-04-07T06:08:55.5898848Z ##[error]Error: The process '/usr/bin/dotnet' failed with exit code 1
2021-04-07T06:08:55.5914545Z Info: Azure Pipelines hosted agents have been updated and now contain .Net 5.x SDK/Runtime along with the older .Net Core version which are currently lts. Unless you have locked down a SDK version for your project(s), 5.x SDK might be picked up which might have breaking behavior as compared to previous versions. You can learn more about the breaking changes here: https://docs.microsoft.com/en-us/dotnet/core/tools/ and https://docs.microsoft.com/en-us/dotnet/core/compatibility/ . To learn about more such changes and troubleshoot, refer here: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/dotnet-core-cli?view=azure-devops#troubleshooting

**Comment by BethanyZhou** [comment]
Hi @Sandido , this error is caused by the ADO environment, which is a random failure.
```
  Segmentation fault (core dumped)
/home/vsts/work/1/s/build.proj(182,5): error MSB3073: The command "dotnet sln /home/vsts/work/1/s/artifacts/Azure.PowerShell.sln add "/home/vsts/work/1/s/src/RecoveryServices/RecoveryServices.Backup.Providers/RecoveryServices.Backup.Providers.csproj"" exited with code 139.
```
Let's re-run all pipelines.

**Comment by erich-wang** [comment]
Force merge due to known CI issue

**Pull request merged**